Source/WebCore/dom/StaticRange.cpp

```diff
@@ -83,4 +83,12 @@ void StaticRange::visitNodesConcurrently(JSC::AbstractSlotVisitor& visitor) cons
     addWebCoreOpaqueRoot(visitor, end.container.get());
 }
 
+bool StaticRange::isValid() const
+{
+    if ((endOffset() < startOffset() && startContainer().isSameNode(&endContainer())) || collapsed() || startContainer().isDocumentNode() || endContainer().isDocumentNode() || startContainer().isDocumentTypeNode() || endContainer().isDocumentTypeNode()
+        || !connectedInSameTreeScope(&startContainer().rootNode(), &endContainer().rootNode())
+        || is_gt(treeOrder<ComposedTree>(startContainer(), endContainer())))
```

Comment:
Hmm... we should probably reorganize this section

Reply:
How about something like this?

```cpp
bool StaticRange::isValid() const
{
    if (collapsed())
        return false;
    if (endOffset() < startOffset() && startContainer().isSameNode(&endContainer()))
        return false;
    if (startContainer().isDocumentNode() || startContainer().isDocumentTypeNode())
        return false;
    if (endContainer().isDocumentNode() || endContainer().isDocumentTypeNode())
        return false;
    if (!connectedInSameTreeScope(&startContainer().rootNode(), &endContainer().rootNode()))
        return false;
    if (is_gt(treeOrder<ComposedTree>(startContainer(), endContainer())))
        return false;
    return endOffset() <= endContainer().length() && startOffset() <= startContainer().length();
}
```

Reply:
For what it's worth, I find @aprotyas 's proposal more readable too.

Reply:
will do 👍

Reply:
you could also probably cache `endContainer` and `startContainer` into local variables